arwen/contexts/runtime.go

```diff
@@ -688,10 +689,14 @@ func (context *runtimeContext) MemLoad(offset int32, length int32) ([]byte, erro
 	memory := context.instance.InstanceCtx.Memory()
 	memoryView := memory.Data()
 	memoryLength := memory.Length()
-	requestedEnd := uint32(offset + length)
+	requestedEnd, err := math.AddInt32(offset, length)
```

Comment:
Use a variant of `AddInt32` which simply returns MaxInt32 if the addition overflows, and no error. The condition at 702 makes it safe.

Reply:
Done.